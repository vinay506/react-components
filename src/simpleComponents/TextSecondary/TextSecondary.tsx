import React from 'react'
import styled from 'styled-components';
function TextSecondary(props: any) {
    /**
     * Text: The text which should be displayed on web page
     * color: Defaults to black, To override the existing color we can pass Color code or color name as props.
     */
    const { text, styles } = props;
    
    return (
        <React.Fragment>
            <Label id="textSecondary" style={styles}>
                {text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    font-size: 18px;
    color: black;
    font-family: 'AmazonEmber-bold'
    `;

export default TextSecondary
