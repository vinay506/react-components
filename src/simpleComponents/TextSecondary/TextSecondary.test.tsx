import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import TextSecondary from './TextSecondary';
import 'jest-styled-components';
describe('Text Secondary', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    beforeEach(() => {
        propInfo = {
            text: 'Sample Text Secondary',
            styles: {
                color: 'red'
            }
        };
        defaultStyles = {
            color: 'black',
            fontSize: '18px',
            fontFamily: 'AmazonEmber-bold',
            letterSpacing: '0',
            lineHeight: '24px'
        }
        wrapper = shallow(<TextSecondary {...propInfo} />);
    });
    test('Secondary Text Snapshot', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Secondary Text Element check', () => {
        const elementLength = wrapper.find('#textSecondary').length;
        expect(elementLength).toEqual(1);
    })
    test('Secondary Text Validation', () => {
        const primaryText = wrapper.find('#textSecondary').text();
        expect(primaryText).toEqual(propInfo.text);
    })
    test('Secondary Text Color Validation', () => {
        const elementStyles = wrapper.find('#textSecondary');
        expect(elementStyles.props().style.color).toEqual(propInfo.styles.color);
    })

    test('Text Secondary Default Color Validation', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Text Secondary Default font-size Validation', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Text Secondary Default font-family Validation', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontFamily);
    })

    test('Text Secondary Default letter Spacing Validation', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.letterSpacing);
    })

    test('Text Secondary Default line Height Validation', () => {
        const instance = renderer.create(<TextSecondary {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.lineHeight);
    })
})

