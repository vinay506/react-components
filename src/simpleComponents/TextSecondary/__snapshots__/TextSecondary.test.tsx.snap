// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Secondary Secondary Text Snapshot 1`] = `
.c0 {
  font-size: 18px;
  color: black;
  font-family: 'AmazonEmber-bold';
}

<div
  className="c0"
  id="textSecondary"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text Secondary
</div>
`;
